### PR TITLE
Configurable SDP Track Count

### PR DESCRIPTION
**Pull request by sirknightj**
*Issue #, if available:*
- N/A

*What was changed?*
- Made `MAX_SDP_SESSION_MEDIA_COUNT` a CMake-configurable parameter (default: 5/unchanged, range: 1-6) instead of a hardcoded value.
- Replaced `CHK` with `CHK_ERR` at all media count limit checks for observability; to log the max and current count on failure.
- Added `docs/SDP_MEDIA_COUNT.md` documenting what consumes slots, stack size implications, and troubleshooting.

*Why was it changed?*
- Adding support for 4 video track + 1 audio track + data channel use case, which consumes 6 slots.
- Keeping it backwards compatible for existing users in terms of memory consumption.

*How was it changed?*
- `Sdp.h`: Replaced `#ifdef ENABLE_DATA_CHANNEL` (5 vs 6) with a single `#ifndef MAX_SDP_SESSION_MEDIA_COUNT` guard defaulting to 5.
- `CMakeLists.txt`: Added `MAX_SDP_SESSION_MEDIA_COUNT` cache variable with range validation (1-6) and conditional `add_definitions`.
- `README.md`: Added the new `-DMAX_SDP_SESSION_MEDIA_COUNT=N` flag entry linking to the doc.
- `Deserialize.c`: Replaced `CHK` with `CHK_ERR` to log max and current count on failure.
- Added `docs/SDP_MEDIA_COUNT.md` with configuration, slot breakdown, stack sizing, and troubleshooting.

*What testing was done for the changes?*
- Tested with `MAX_SDP_SESSION_MEDIA_COUNT=6` on macOS: all 53 SDP tests pass.
- Tested with `MAX_SDP_SESSION_MEDIA_COUNT=6` end-to-end against the JS test page via an internal demo.
- Tested with `MAX_SDP_SESSION_MEDIA_COUNT=1` on macOS: the SDK errors gracefully with `STATUS_SESSION_DESCRIPTION_MAX_MEDIA_COUNT` (0x5500000A) and does not crash.
- Verified CMake rejects out-of-range values (0 and 7).
- Unit tests cover deserialization and offer generation boundary conditions at the limit.
- Data channel test validates `MAX_SDP_SESSION_MEDIA_COUNT - 1` transceivers + data channel succeeds, and one more fails.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by YPandas — APPROVED** [review]

File: CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +37,20 @@ option(BUILD_OLD_MBEDTLS_VERSION "Use MbedTLS version 2.28.8." OFF)
 option(INCREASE_PRECISION_TIMING_LOGS "PROFILE-level timing logs have 2 decimals of milliseconds (otherwise truncated to whole millisecond)" ON)
 option(PARALLEL_BUILD "Build dependencies with parallel flag" OFF)
 
+# Tuning
+set(MAX_SDP_SESSION_MEDIA_COUNT "5" CACHE STRING "Maximum number of SDP media sections (m= lines). Range: 1-6. Each slot adds ~140 KB to stack-allocated SessionDescription. See docs/SDP_MEDIA_COUNT.md.")
+if (NOT "${MAX_SDP_SESSION_MEDIA_COUNT}" STREQUAL "5")
```
Comment (Ahhorse1) [inline CMakeLists.txt:45]: Should there be a regex check to ensure MAX_SDP_SESSION_MEDIA_COUNT is a number?

Comment (sirknightj) [reply CMakeLists.txt:45]: Added the regex and verified locally with some test cases:
```
cmake .. -DMAX_SDP_SESSION_MEDIA_COUNT=abc
...
MAX_SDP_SESSION_MEDIA_COUNT must be a number, got 'abc'.
```

Also tested 5 and 6 locally.

**Review by Ahhorse1 — APPROVED** [review]
